diff -urN orig/os_dep/linux/ioctl_cfg80211.c mod/os_dep/linux/ioctl_cfg80211.c
--- orig/os_dep/linux/ioctl_cfg80211.c	2026-08-26 13:47:17.251459920 +0800
+++ mod/os_dep/linux/ioctl_cfg80211.c	2026-08-26 13:47:57.218656283 +0800
@@ -437,7 +437,9 @@
 
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 19, 0))
 	if (started) {
-		#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 11, 0))
+		#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0))
+		cfg80211_ch_switch_started_notify(adapter->pnetdev, &chdef, 0, 0, false);
+		#elif (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 11, 0))
 		cfg80211_ch_switch_started_notify(adapter->pnetdev, &chdef, 0, false);
 		#else
 		cfg80211_ch_switch_started_notify(adapter->pnetdev, &chdef, 0);
@@ -449,7 +451,11 @@
 	if (!rtw_cfg80211_allow_ch_switch_notify(adapter))
 		goto exit;
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0))
+	cfg80211_ch_switch_notify(adapter->pnetdev, &chdef, 0);
+#else
 	cfg80211_ch_switch_notify(adapter->pnetdev, &chdef);
+#endif
 
 #else
 	int freq = rtw_ch2freq(ch);
@@ -948,16 +954,24 @@
 	wdev = a->rtw_wdev;
 	network = &a->mlmeextpriv.mlmext_info.network;
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0))
+	if ((!wdev->u.client.ssid_len) || (wdev->u.client.ssid_len != network->Ssid.SsidLength)
+	    || (_rtw_memcmp(wdev->u.client.ssid, network->Ssid.Ssid,
+			    network->Ssid.SsidLength) == _FALSE)) {
+#else
 	if ((!wdev->ssid_len) || (wdev->ssid_len != network->Ssid.SsidLength)
 	    || (_rtw_memcmp(wdev->ssid, network->Ssid.Ssid,
 			    network->Ssid.SsidLength) == _FALSE)) {
+#endif
 		RTW_PRINT(FUNC_ADPT_FMT ": bssid:"MAC_FMT"\n",
 			  FUNC_ADPT_ARG(a), MAC_ARG(network->MacAddress));
 		RTW_PRINT(FUNC_ADPT_FMT ": ssid:[%s] len=%d\n",
 			  FUNC_ADPT_ARG(a), network->Ssid.Ssid,
 			  network->Ssid.SsidLength);
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 1, 0))
 		RTW_PRINT(FUNC_ADPT_FMT ": (wdev) ssid:[%s] len=%d\n",
 			  FUNC_ADPT_ARG(a), wdev->ssid, wdev->ssid_len);
+#endif
 
 		return _FALSE;
 	}
@@ -1162,7 +1176,12 @@
 		#endif
 
 		#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 12, 0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0)
+		roam_info.links[0].bssid = cur_network->network.MacAddress;
+		roam_info.links[0].channel = notify_channel;
+#else
 		roam_info.bssid = cur_network->network.MacAddress;
+#endif
 		roam_info.req_ie = pmlmepriv->assoc_req + sizeof(struct rtw_ieee80211_hdr_3addr) + 2;
 		roam_info.req_ie_len = pmlmepriv->assoc_req_len - sizeof(struct rtw_ieee80211_hdr_3addr) - 2;
 		roam_info.resp_ie = pmlmepriv->assoc_rsp + sizeof(struct rtw_ieee80211_hdr_3addr) + 6;
@@ -1727,6 +1746,9 @@
 }
 
 static int cfg80211_rtw_add_key(struct wiphy *wiphy, struct net_device *ndev
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0))
+	, int link_id
+#endif
 	, u8 key_index
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 37)) || defined(COMPAT_KERNEL_RELEASE)
 	, bool pairwise
@@ -1871,6 +1893,9 @@
 }
 
 static int cfg80211_rtw_get_key(struct wiphy *wiphy, struct net_device *ndev
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0))
+	, int link_id
+#endif
 	, u8 keyid
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 37)) || defined(COMPAT_KERNEL_RELEASE)
 	, bool pairwise
@@ -2036,6 +2061,9 @@
 }
 
 static int cfg80211_rtw_del_key(struct wiphy *wiphy, struct net_device *ndev,
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0))
+				int link_id,
+#endif
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 37)) || defined(COMPAT_KERNEL_RELEASE)
 				u8 key_index, bool pairwise, const u8 *mac_addr)
 #else	/* (LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 37)) */
@@ -2056,7 +2084,11 @@
 }
 
 static int cfg80211_rtw_set_default_key(struct wiphy *wiphy,
-	struct net_device *ndev, u8 key_index
+	struct net_device *ndev,
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0))
+	int link_id,
+#endif
+	u8 key_index
 	#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 38)) || defined(COMPAT_KERNEL_RELEASE)
 	, bool unicast, bool multicast
 	#endif
@@ -2104,7 +2136,11 @@
 
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 30))
 int cfg80211_rtw_set_default_mgmt_key(struct wiphy *wiphy,
-	struct net_device *ndev, u8 key_index)
+	struct net_device *ndev,
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0))
+	int link_id,
+#endif
+	u8 key_index)
 {
 #define SET_DEF_KEY_PARAM_FMT " key_index=%d"
 #define SET_DEF_KEY_PARAM_ARG , key_index
@@ -5330,7 +5366,11 @@
 	return ret;
 }
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0))
+static int cfg80211_rtw_stop_ap(struct wiphy *wiphy, struct net_device *ndev, unsigned int link_id)
+#else
 static int cfg80211_rtw_stop_ap(struct wiphy *wiphy, struct net_device *ndev)
+#endif
 {
 	_adapter *adapter = (_adapter *)rtw_netdev_priv(ndev);
 
@@ -10452,7 +10492,8 @@
 
 	rtw_cfg80211_indicate_scan_done(adapter, _TRUE);
 
-	#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 11, 0)) || defined(COMPAT_KERNEL_RELEASE)
+	#if ((LINUX_VERSION_CODE >= KERNEL_VERSION(3, 11, 0)) || defined(COMPAT_KERNEL_RELEASE)) && \
+		(LINUX_VERSION_CODE < KERNEL_VERSION(6, 1, 0))
 	if (wdev->current_bss) {
 		RTW_INFO(FUNC_ADPT_FMT" clear current_bss by cfg80211_disconnected\n", FUNC_ADPT_ARG(adapter));
 		rtw_cfg80211_indicate_disconnect(adapter, 0, 1);
diff -urN orig/os_dep/linux/rtw_proc.c mod/os_dep/linux/rtw_proc.c
--- orig/os_dep/linux/rtw_proc.c	2026-08-26 13:47:17.252908020 +0800
+++ mod/os_dep/linux/rtw_proc.c	2026-08-26 13:47:57.222656309 +0800
@@ -37,6 +37,8 @@
 #if (LINUX_VERSION_CODE < KERNEL_VERSION(3, 10, 0))
 #define PDE_DATA(inode) PDE((inode))->data
 #define proc_get_parent_data(inode) PDE((inode))->parent->data
+#elif (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 17, 0))
+#define PDE_DATA(inode) pde_data(inode)
 #endif
 
 #if (LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 24))
